core/Property/PropertySetting.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Property;
+
+use Piwik\Piwik;
+
+/**
+ * Describes a Type setting for a website, mobile app, ...
+ *
+ * See {@link \Piwik\Plugin\Settings}.
+ */
+class PropertySetting extends \Piwik\Settings\Setting
+{
+    /**
+     * By default the value of the type setting is only readable by users having at least view access to one site
+     *
+     * @var bool
+     * @since 2.14.0
+     */
+    public $readableByCurrentUser = false;
+
+    /**
+     * By default the value of the type setting is only writable by users having at least admin access to one site
+     * @var bool
+     * @internal
+     */
+    public $writableByCurrentUser = false;
+
+    /**
+     * Constructor.
+     *
+     * @param string $name The persisted name of the setting.
+     * @param string $title The display name of the setting.
+     */
+    public function __construct($name, $title)
+    {
+        parent::__construct($name, $title);
+
+        $this->writableByCurrentUser = Piwik::isUserHasSomeAdminAccess();
+        $this->readableByCurrentUser = Piwik::isUserHasSomeViewAccess();
+    }
```

Comment:
At this time we do not yet have a `idSite` which is kinda a trade off so here we only check for general permissions. Before saving we actually check whether a user has admin access for a selected site. In general we always check for permissions when reading the value anyway.